iac-extensions/kubernetes/src/main/resources/org/sonar/l10n/kubernetes/rules/kubernetes/S117.html

```diff
@@ -0,0 +1,68 @@
+maintain and debug. It also ensures consistency in the code, especially when multiple developers are working on the same project.</p>
+<p>This rule checks that local variable and function parameter names match a provided regular expression.</p>
+<h3>What is the potential impact?</h3>
+<p>Inconsistent naming of local variables and function parameters can lead to several issues in your code:</p>
+<ul>
+  <li> <strong>Reduced Readability</strong>: Inconsistent local variable and function parameter names make the code harder to read and understand;
+  consequently, it is more difficult to identify the purpose of each variable, spot errors, or comprehend the logic. </li>
+  <li> <strong>Difficulty in Identifying Variables</strong>: The local variables and function parameters that don’t adhere to a standard naming
+  convention are challenging to identify; thus, the coding process slows down, especially when dealing with a large codebase. </li>
+  <li> <strong>Increased Risk of Errors</strong>: Inconsistent or unclear local variable and function parameter names lead to misunderstandings about
+  what the variable represents. This ambiguity leads to incorrect assumptions and, consequently, bugs in the code. </li>
+  <li> <strong>Collaboration Difficulties</strong>: In a team setting, inconsistent naming conventions lead to confusion and miscommunication among
+  team members. </li>
+  <li> <strong>Difficulty in Code Maintenance</strong>: Inconsistent naming leads to an inconsistent codebase. The code is difficult to understand,
+  and making changes feels like refactoring constantly, as you face different naming methods. Ultimately, it makes the codebase harder to maintain.
+  </li>
+</ul>
+<p>In summary, not adhering to a naming convention for local variables and function parameters can lead to confusion, errors, and inefficiencies,
+making the code harder to read, understand, and maintain.</p>
+<h2>How to fix it</h2>
+<p>First, familiarize yourself with the particular naming convention of the project in question. Then, update the name to match the convention, as
+well as all usages of the name. For many IDEs, you can use built-in renaming and refactoring features to update all usages at once.</p>
+<p>As a default, we suggest using the camelCase naming convention, as this is widely adopted for variables in Helm.</p>
+<h3>Code examples</h3>
+<h4>Noncompliant code example</h4>
+<p>With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
+<pre data-diff-id="1" data-diff-type="noncompliant">
+{{- $my_Variable := .Values.myVariableValue -}} # Noncompliant
+</pre>
+<pre data-diff-id="2" data-diff-type="noncompliant">
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-configmap
+data:
+  myvalue: "Hello World"
+  {{- range $KeyNc, $val-nc := .Values.favorite }} # Noncompliant
```

Comment:
Variables in Go templates cannot be in `kebab-case`; this should be changed in the rpsec